joss/paper.md

```diff
@@ -96,17 +96,49 @@ Basix allows users to:
-and add new elements to Basix without needing to make changes to the rest
-of the library. This also allows users who want to create custom
+- define custom finite elements.
+
+In order to compute the basis functions of a finite element, each functional
+in $\mathcal{L}$ can be applied to the elements of a basis of the polynomial
+set $\mathcal{V}$ to compute the *dual matrix*. This dual matrix can then be
+inverted to obtain the coefficients that define the finite element basis
+functions in terms of the basis of $\mathcal{V}$. In NGSolve [@ngsolve], the
+functionals for this approach are implemented in C++ using lambda functions.
+In MFEM [@mfem], the functionals are described in C++ using a set of points and
+weights: for point evaluation functionals, there will be one point and the
+weight will be 1; for integral functions, quadrature points and weights are used.
+In FIAT [@fiat] and Symfem [@symfem], the functionals are implemented as
+Python objects. In Basix, we follow an approach similar to that taken my MFEM,
+and define the functionals in C++ using a set of points and weights.
+
+In many other libraries---including Dune [@dune] and Bempp [@bempp; @bempp-cl]---the
+basis functions of the finite element space are implemented explicitly.
+
+In a large number of finite element libraries---including NGSolve and MFEM---the
+finite element definitions are included in the core of the library. In Dune,
+the finite element definitions are separated into the `dune-localfunctions` module.
+FIAT and Basix are the finite element definition and tabulations libraries used
+by legacy FEniCS and FEniCSx (respectively). Symfem is a standalone element definition
+libraries that computed basis functions symbolically. By separating the finite element
+definitions from the core of the code, users are able to tabulate elements and
+obtain information about them without having to interact with the other components of the library.
+In FEniCSx for example, this allows users who want to create custom
 integration kernels to get information about elements from Basix without
 having to extract information from the core of the full finite element
 library.
 
+An additional advantage of this modular approach is that it allows
+us to adjust how elements are implemented
+and add new elements without needing to make changes to the other components
+of FEniCSx.
+
+For some high order finite elements on tensor product cells (quadrilaterals and hexahedra),
+elements can be more efficiently tabulated using sum factorisation: the elements
+can be represented as the product of elements defined on an interval, and so can be
+tabulated by combining copies of tabulated data on an interval rather than tabulating on the full cell.
+This tensor product evaluation is implemented in FInAT [@finat1; @finat2] and deal.ii [@dealii].
+We have experimented with supporting these factorisations in Basix, and plan to support them more fully in a future release.
```

Comment:
Maybe we can mention that for some elements (Lagrange, for instance) it's already possible to obtain the factors (subelements) to recreate the basis functions of the full element. But how to combine this factors to compute derivatives is still a WIP, and currently left to the user.
But on second thought, we don't need to go into detail, since it's a work in progress.